piptools/repositories/local.py

```diff
@@ -72,12 +72,9 @@ def find_best_match(
         existing_pin = self.existing_pins.get(key)
         if existing_pin and ireq_satisfied_by_existing_pin(ireq, existing_pin):
             try:
-                self.repository.find_best_match(existing_pin, prereleases)
+                return self.repository.find_best_match(existing_pin, prereleases)
```

Comment:
Oh, I'm now doubting this part, as maybe info from the original ireq gets lost along the way.